src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ManageContent/ReleaseContentBlockService.cs

```diff
@@ -0,0 +1,213 @@
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.EntityFrameworkCore;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services.ManageContent
+{
+    public class ReleaseContentBlockService : IReleaseContentBlockService
+    {
+        private readonly ContentDbContext _contentDbContext;
+        private readonly IPersistenceHelper<ContentDbContext> _persistenceHelper;
+        private readonly IUserService _userService;
+        private readonly IHubContext<ReleaseContentHub, IReleaseContentHubClient> _hubContext;
+
+        public ReleaseContentBlockService(
+            ContentDbContext contentDbContext,
+            IPersistenceHelper<ContentDbContext> persistenceHelper,
+            IUserService userService,
+            IHubContext<ReleaseContentHub, IReleaseContentHubClient> hubContext)
+        {
+            _contentDbContext = contentDbContext;
+            _persistenceHelper = persistenceHelper;
+            _userService = userService;
+            _hubContext = hubContext;
+        }
+
+        /// <summary>
+        /// Try to acquire the lock for a content block, notifying any
+        /// subscribed hub clients if the lock was acquired.
+        /// </summary>
+        /// <returns>
+        /// The lock details if was acquired. If another user
+        /// holds the lock, then we return that instead.
+        /// </returns>
+        public async Task<Either<ActionResult, ReleaseContentBlockLockViewModel>> LockContentBlock(
+            Guid id,
+            bool force = false)
+        {
+            var userId = _userService.GetUserId();
+
+            return await GetContentBlock(id)
+                .OnSuccessDo(CheckCanUpdateBlock)
+                .OnSuccess(
+                    async block =>
+                    {
+                        if (TryFindExistingLock(block, userId, force, out var existingLock))
+                        {
+                            return existingLock;
+                        }
+
+                        return await DoContentBlockLock(block, userId);
+                    }
+                );
+        }
+
+        private async Task<ReleaseContentBlockLockViewModel> DoContentBlockLock(ContentBlock block, Guid userId)
+        {
+            var user = await _contentDbContext.Users.FindAsync(userId);
+
+            if (user is null)
+            {
+                throw new ArgumentException($"User with id {userId} does not exist", nameof(userId));
+            }
+
+            _contentDbContext.ContentBlocks.Update(block);
```

Comment:
Why the immediate update here plz?  I think most places we tend to call this after we've made amendments to the entity in question, even if this is just a code style choice.  Wondering if this was needed for another tracking purpose?

Reply:
Mainly placed these beforehand because I've seen this done in other places by @benoutram. I don't think it really matters either way as it seems that `Update` only really toggles the entity's persistence state to `Modified`.

I agree that it probably makes more sense to place these after any modifications so I've moved the `Update` call as suggested.